### PR TITLE
Add store

### PR DESCRIPTION
**Pull request by IrinaSendetskaya**

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,20 +1,35 @@
-<!--The content below is only a placeholder and can be replaced.-->
-<div style="text-align:center">
-  <h1>
-    Welcome to {{ title }}!
-  </h1>
-  <img width="300" alt="Angular Logo" src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNTAgMjUwIj4KICAgIDxwYXRoIGZpbGw9IiNERDAwMzEiIGQ9Ik0xMjUgMzBMMzEuOSA2My4ybDE0LjIgMTIzLjFMMTI1IDIzMGw3OC45LTQzLjcgMTQuMi0xMjMuMXoiIC8+CiAgICA8cGF0aCBmaWxsPSIjQzMwMDJGIiBkPSJNMTI1IDMwdjIyLjItLjFWMjMwbDc4LjktNDMuNyAxNC4yLTEyMy4xTDEyNSAzMHoiIC8+CiAgICA8cGF0aCAgZmlsbD0iI0ZGRkZGRiIgZD0iTTEyNSA1Mi4xTDY2LjggMTgyLjZoMjEuN2wxMS43LTI5LjJoNDkuNGwxMS43IDI5LjJIMTgzTDEyNSA1Mi4xem0xNyA4My4zaC0zNGwxNy00MC45IDE3IDQwLjl6IiAvPgogIDwvc3ZnPg==">
+<div class="header__navbar">
+  <nav>
+    <img class="header__navbar_item" src="../assets/images/logo.png" />
+    <a class="header__navbar_item">
```
Comment (nastabulak) [inline src/app/app.component.html]: better to have separate header component

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,71 @@
...
+  }
+  findAllUsers() {
+    this.findUserSubscription = this.userService
+      .getAllUsers()
+      .subscribe(data => {
+        this.users = data["users"];
+        this.store$.dispatch(new LoadUsers(this.users));
+      });
+  }
+  unsubscribe(subscription: Subscription) {
```
Comment (nastabulak) [inline src/app/app.component.ts]: better use takeUntil

File: src/app/components/add-course/add-course.component.html [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  <div class="form-group">
+    <button
+      [disabled]="nameCourse.invalid || description.invalid || duration.invalid"
+      class="button buttonSubmit"
+      (click)="addNewCourse()"
+    >
+      Сохранить
+    </button>
+
+    <button class="button buttonCancel" (click)="cancelAddNewCourse()">
```
Comment (nastabulak) [inline src/app/components/add-course/add-course.component.html]: class name is not according your class namiing

File: src/app/components/breadcrumb/breadcrumb.component.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      label: label,
+      url: nextUrl
+    };
+    const newBreadcrumbs = breadcrumb.label
+      ? [...breadcrumbs, breadcrumb]
+      : [...breadcrumbs];
+    return newBreadcrumbs;
+  }
+
+  findLabelIFDynamicPathAndReturnBreadcrumb(
```
Comment (nastabulak) [inline src/app/components/breadcrumb/breadcrumb.component.ts]: check spelling)

File: src/app/components/courses/courses.component.css [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    flex-direction: row;
+    justify-content: space-around;
+    align-items: flex-start;
+    padding: 5px;
+  }
+  .list__course__item__description {
+    margin-right: 20px;
+    justify-self: stretch;
+  }
+  .course__item__description__header__date {
```
Comment (nastabulak) [inline src/app/components/courses/courses.component.css]: revise BEM

File: src/app/components/datepicker/datepicker.component.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<mat-form-field>
+  <input matInput [matDatepicker]="dp" placeholder="dd.MM.yyyy" [formControl] ="date" [value]="dateForParent">
+  <mat-datepicker-toggle matSuffix [for]="dp"></mat-datepicker-toggle>
+  <mat-datepicker #dp ></mat-datepicker>
```
Comment (nastabulak) [inline src/app/components/datepicker/datepicker.component.html]: please avoid cutting of words for clearer code

File: src/app/components/datepicker/datepicker.component.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  ]
+})
+export class DatepickerComponent implements OnChanges {
+  @Input() dateForParent: Date;
+  @Output() dateOut: EventEmitter<Date> = new EventEmitter<Date>();
+
+  date = new FormControl(moment(),Validators.required);
+
+  ngOnChanges() {
+    this.date.valueChanges.subscribe(data => {
```
Comment (nastabulak) [inline src/app/components/datepicker/datepicker.component.ts]: subscribe and no unsubscribe

File: src/app/shared/duration.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+@Pipe({
+  name: 'duration'
+})
+export class DurationPipe implements PipeTransform {
+
+  transform(value: number, args?: any): string {
+
+    let result='';
+    if(value<=0) return ' час  мин';
```
Comment (nastabulak) [inline src/app/shared/duration.pipe.ts]: no formatting

File: src/app/components/datepicker/datepicker.component.html (hunk 6 above)
Comment (IrinaSendetskaya) [reply src/app/components/datepicker/datepicker.component.html]: changed

File: src/app/components/datepicker/datepicker.component.ts (hunk 7 above)
Comment (IrinaSendetskaya) [reply src/app/components/datepicker/datepicker.component.ts]: changed

File: src/app/app.component.html (hunk 1 above)
Comment (IrinaSendetskaya) [reply src/app/app.component.html]: changed

File: src/app/app.component.ts (hunk 2 above)
Comment (IrinaSendetskaya) [reply src/app/app.component.ts]: changed

File: src/app/components/add-course/add-course.component.html (hunk 3 above)
Comment (IrinaSendetskaya) [reply src/app/components/add-course/add-course.component.html]: changed

File: src/app/components/breadcrumb/breadcrumb.component.ts (hunk 4 above)
Comment (IrinaSendetskaya) [reply src/app/components/breadcrumb/breadcrumb.component.ts]: changed

File: src/app/components/courses/courses.component.css (hunk 5 above)
Comment (IrinaSendetskaya) [reply src/app/components/courses/courses.component.css]: changed

File: src/app/shared/duration.pipe.ts (hunk 8 above)
Comment (IrinaSendetskaya) [reply src/app/shared/duration.pipe.ts]: changed

**Pull request merged**